hack/make/validate-vendor

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+source "${MAKEDIR}/.validate"
+
+IFS=$'\n'
+files=( $(validate_diff --diff-filter=ACMR --name-only -- 'hack/vendor.sh' || true) )
+unset IFS
+
+badFiles=()
+for f in "${files[@]}"; do
+	# We run vendor.sh to and see if we have a diff afterwards
+	./hack/vendor.sh
+	# Let see if the working directory is clean
+	diffs=( $(git status --porcelain 2>/dev/null| egrep "^( M|A|D)") )
```

Comment:
hmm, this is actually going to have false-positives if we've got other changes in our local copy -- maybe just `git status --porcelain -- vendor` to make sure it's limited to just the vendor directory's changes?  Then, you could store it in a string instead, and check whether the string is empty, and then use the string as the error message output telling us where the problems are. :smile: